### PR TITLE
Finished svc doc creation

### PR DESCRIPTION
**Pull request by zasamen**

### CODE REVIEW

**Comment by zasamen** [comment]
Updated.

File: server/src/main/java/school/journal/controller/ClassAPIController.java [hunk 1]
```diff
@@ -76,14 +77,22 @@ public ResponseEntity getOne(HttpServletRequest request, @PathVariable("classId"
...
             throws ControllerException {
         try {
-//            generationService.generateMarksDocument(DocumentType.CSV, 1, 1);
-            generationService.generateClassPupilListDocument(response.getOutputStream(), DocumentType.PDF, 1);
-        } catch (ServiceException | IOException exc) {}
+            response.setContentType("text/csv");
+            response.setHeader("Content-Disposition", "attachment;filename=mdk.csv");
+//            response.setContentType("text/pdf");
+
+//            generationService.generateMarksDocument(response.getOutputStream(), DocumentType.CSV, 1, 1);
```
Comment (mbarouski) [inline server/src/main/java/school/journal/controller/ClassAPIController.java]: what are comments there?

File: server/src/main/java/school/journal/service/IDocumentGenerationService.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
+package school.journal.service;
```
Comment (mbarouski) [inline server/src/main/java/school/journal/service/IDocumentGenerationService.java]: what is it?

File: server/src/main/java/school/journal/service/document/generation/CSVService.java [hunk 3]
```diff
@@ -0,0 +1,481 @@
...
+    public OutputStream generateMarksDocument(OutputStream os, Subject subject, List<Mark> markList, List<Pupil> pupilList, Clazz clazz, List<Date> lessonDateList) throws ServiceException {
+        CSVWriter writer = createSimpleWriter(os);
+
+        writer.writeNext(wrapMarkListHeader(clazz, subject));
+        writer.writeAll(getMarkTable(pupilList, markList, lessonDateList));
+
+        flush(writer);
+        return os;
+    }
+}
```
Comment (mbarouski) [inline server/src/main/java/school/journal/service/document/generation/CSVService.java]: SHIT!!!!

File: server/src/main/java/school/journal/service/document/generation/GenerationService.java [hunk 4]
```diff
@@ -22,59 +22,53 @@
...
         Clazz clazz = classService.getOne(classId);
         List<Pupil> pupilList = pupilService.getListOfPupils(classId);
-        List<Teacher> teacherList = teacherService.getListOfTeachersForClass(classId);
-        Teacher teacher = null;
-        for (Teacher t : teacherList) {
-            if(t.getClassId() == classId){
-                teacher = t;
-            }
-        }
+//        List<Teacher> teacherList = teacherService.getListOfTeachersForClass(classId);
```
Comment (mbarouski) [inline server/src/main/java/school/journal/service/document/generation/GenerationService.java]: delete

**Review by mbarouski — CHANGES_REQUESTED** [review]

File: server/src/main/java/school/journal/controller/ClassAPIController.java (hunk 1 above)
Comment (zasamen) [reply server/src/main/java/school/journal/controller/ClassAPIController.java]: It is testing endpoint. If u want to try run my generation way, you shouldn't write it.

I checked correctness of generation csv here.

File: server/src/main/java/school/journal/service/document/generation/CSVService.java (hunk 3 above)
Comment (zasamen) [reply server/src/main/java/school/journal/service/document/generation/CSVService.java]: But what do u want?
We shouldn't return output stream.
the best way is null.

File: server/src/main/java/school/journal/service/document/generation/GenerationService.java (hunk 4 above)
Comment (zasamen) [reply server/src/main/java/school/journal/service/document/generation/GenerationService.java]: ok

File: server/src/main/java/school/journal/service/IDocumentGenerationService.java (hunk 2 above)
Comment (zasamen) [reply server/src/main/java/school/journal/service/IDocumentGenerationService.java]: deleted

File: server/src/main/java/school/journal/controller/ClassAPIController.java (hunk 1 above)
Comment (zasamen) [reply server/src/main/java/school/journal/controller/ClassAPIController.java]: ok.deleting

**Pull request merged**